### PR TITLE
Show call result status for the appointment in `ContactPatientBottomSheet` - 2

### PR DESCRIPTION
**Pull request by janhavisinghh**
https://app.shortcut.com/simpledotorg/story/8168/show-call-result-status-for-the-appointment-in-contactpatientbottomsheet

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 ### Internal
 
 - Bump Sentry to v5.7.4  
-- [In Progress: 11 May 2022] Show call result status for the appointment in `ContactPatientBottomSheet`
+- [In Progress: 16 May 2022] Show call result status for the appointment in `ContactPatientBottomSheet`
```
Comment (msasikanth) [inline CHANGELOG.md:8]: nit: there is an extra space at end, can we format the file, please?

File: app/src/test/java/org/simple/clinic/contactpatient/ContactPatientUpdateTest.kt [hunk 2]
```diff
@@ -88,12 +88,12 @@ class ContactPatientUpdateTest {
         .whenEvent(OverdueAppointmentLoaded(appointment))
         .then(assertThatNext(
             hasModel(defaultModel.overdueAppointmentLoaded(appointment)),
-            hasNoEffects()
+            hasEffects(LoadCallResultForAppointment(overdueAppointment.uuid))
```
Comment (msasikanth) [inline app/src/test/java/org/simple/clinic/contactpatient/ContactPatientUpdateTest.kt:91]: I think this is something we need to take a look at later. Whether to load them in a blocking manner one after the other or do them in a single call, both the appointment and call result. For now we can go ahead with this 👍🏾

File: app/src/main/java/org/simple/clinic/contactpatient/ContactPatientUiRenderer.kt [hunk 3]
```diff
@@ -31,6 +32,13 @@ class ContactPatientUiRenderer(
     }
   }
 
+  private fun renderCallResultOutcome(model: ContactPatientModel) {
+    if (model.hasCallResult) {
+      val callResult = model.callResult!!.get()
+      ui.showCallResult(callResult.outcome, callResult.timestamps.updatedAt)
+    }
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/contactpatient/ContactPatientUiRenderer.kt:39]: Will we raise another PR for hiding the call result?

**Review by msasikanth — CHANGES_REQUESTED** [review]
See inline comments

File: app/src/main/java/org/simple/clinic/contactpatient/ContactPatientUiRenderer.kt (hunk 3 above)
Comment (janhavisinghh) [reply app/src/main/java/org/simple/clinic/contactpatient/ContactPatientUiRenderer.kt:41]: resolved

**Review by msasikanth — DISMISSED** [review]

File: app/src/main/java/org/simple/clinic/contactpatient/ContactPatientUpdate.kt [hunk 4]
```diff
@@ -56,7 +56,7 @@ class ContactPatientUpdate(
           .contactPatientInfoLoaded()
     }
 
-    return next(updatedModel)
+    return next(updatedModel, LoadCallResultForAppointment(event.overdueAppointment.get().uuid))
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/contactpatient/ContactPatientUpdate.kt:59]: Overdue appointment here is of type `Optional`, would this cause any issues for us? Are we sure we will have overdue appointment here always?

Comment (janhavisinghh) [reply app/src/main/java/org/simple/clinic/contactpatient/ContactPatientUpdate.kt]: umm good catch. Added a commit for this

**Review by msasikanth — APPROVED** [review]

**Pull request merged**